plugins/modules/foreman_operating_system.py

```diff
@@ -192,81 +184,80 @@ def main():
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    try:
-        # Try to find the Operating System to work on
-        # name is however not unique, but description is, as well as "<name> <major>[.<minor>]"
-        entity = None
-        # If we have a description, search for it
-        if 'description' in operating_system_dict and operating_system_dict['description'] != '':
-            entity = find_operating_system_by_title(module, title=operating_system_dict['description'], failsafe=True)
-        # If we did not yet find a unique OS, search by name & version
-        if entity is None:
-            search_dict = {'name': operating_system_dict['name']}
-            if 'major' in operating_system_dict:
-                search_dict['major'] = operating_system_dict['major']
-            if 'minor' in operating_system_dict:
-                search_dict['minor'] = operating_system_dict['minor']
-            entities = find_entities(OperatingSystem, **search_dict)
-            if len(entities) == 1:
-                entity = entities[0]
-    except Exception as e:
-        module.fail_json(msg='Failed to find entity: %s ' % e)
+    # Try to find the Operating System to work on
+    # name is however not unique, but description is, as well as "<name> <major>[.<minor>]"
+    entity = None
+    # If we have a description, search for it
+    if 'description' in entity_dict and entity_dict['description'] != '':
+        entity = module.find_resource_by_title('operatingsystems', entity_dict['description'], failsafe=True)
+    # If we did not yet find a unique OS, search by name & version
+    if entity is None:
+        search_parts = ['{}="{}"'.format(key, entity_dict[key]) for key in ('name', 'major', 'minor') if key in entity_dict]
+        entity = module.find_resource('operatingsystems', ','.join(search_parts), failsafe=True)
 
-    if not entity and (state == 'present' or state == 'present_with_defaults'):
+    if not entity and (module.state == 'present' or module.state == 'present_with_defaults'):
         # we actually attempt to create a new one...
         for param_name in ['major', 'family', 'password_hash']:
-            if param_name not in operating_system_dict.keys():
+            if param_name not in entity_dict.keys():
                 module.fail_json(msg='{} is a required parameter to create a new operating system.'.format(param_name))
 
-    # Set Architectures of Operating System
-    if 'architectures' in operating_system_dict:
-        operating_system_dict['architectures'] = find_entities_by_name(
-            Architecture, operating_system_dict['architectures'], module)
-
-    # Set Installation Media of Operating System
-    if 'media' in operating_system_dict:
-        operating_system_dict['media'] = find_entities_by_name(
-            Media, operating_system_dict['media'], module)
+    if not module.desired_absent:
+        # Set Architectures of Operating System
```

Comment:
I find these comments (here and in the following assignments) quite redundant to the actual code, so I'd drop them (but up to you if you'd do that ;))